tests/Unit/Http/Action/Security/LogInActionTest.php

```diff
@@ -109,10 +110,29 @@ public function rendersLogInFormIfAuthenticationFailed()
         $this->assertSame($content, $response->getContent());
     }
 
+    /**
+     * Data provider for redirectsToDashboardIfAuthenticationSucceeded
+     *
+     * @return array
+     */
+    public function userProvider(): array
+    {
+        return [ // admin check, reviewer check, expected redirect
+            [false, false, 'dashboard'],
+            [false, true, 'reviewer'],
+            [true, null, 'admin'],
+        ];
+    }
+
     /**
      * @test
+     *
+     * @param bool      $isAdmin
+     * @param null|bool $isReviewer
+     * @param string    $shouldRedirectTo
+     * @dataProvider    userProvider
      */
-    public function redirectsToDashboardIfAuthenticationSucceeded()
+    public function redirectsToDashboardIfAuthenticationSucceeded($isAdmin, $isReviewer, $shouldRedirectTo)
```

Comment:
Can we add type and return-type declarations here?

Reply:
Done.